frontend/controller/actions/user.js

```diff
@@ -0,0 +1,102 @@
+import sbp from '~/shared/sbp.js'
+import { GIErrorUIRuntimeError } from '@model/errors.js'
+import L from '@view-utils/translations.js'
+
+// import { CONTRACT_IS_SYNCING } from '@utils/events.js'
+
+export default sbp('sbp/selectors/register', {
+  'gi.actions/user/create': async function ({
+    username,
+    email,
+    password
+  }) {
+    // TODO: make sure we namespace these names:
+    //       https://github.com/okTurtles/group-income-simple/issues/598
+    const oldSettings = await sbp('gi.db/settings/load', username)
+    if (oldSettings) {
+      // TODO: prompt to ask user before deleting and overwriting an existing user
+      //       https://github.com/okTurtles/group-income-simple/issues/599
+      console.warn(`deleting settings for pre-existing user ${username}!`, oldSettings)
+      await sbp('gi.db/settings/delete', username)
+    }
+    // proceed with creation
+    const user = sbp('gi.contracts/identity/create', {
+      // authorizations: [Events.CanModifyAuths.dummyAuth()],
+      attributes: {
+        name: username,
+        email: email,
+        picture: `${window.location.origin}/assets/images/default-avatar.png`
+      }
+    })
+    const mailbox = sbp('gi.contracts/mailbox/create', {
+      // authorizations: [Events.CanModifyAuths.dummyAuth(user.hash())]
+    })
+    await sbp('backend/publishLogEntry', user)
+    await sbp('backend/publishLogEntry', mailbox)
+
+    return [user.hash(), mailbox.hash()]
+  },
+  'gi.actions/user/signup': async function ({
+    username,
+    email,
+    password // TODO - implement
+  }, {
+    sync = true
+  } = {}) {
+    try {
+      const [userID, mailboxID] = await sbp('gi.actions/user/create', { username, email, password })
+
+      // set the attribute *after* publishing the identity contract
+      const attribute = await sbp('gi.contracts/identity/setAttributes/create',
+        { mailbox: mailboxID },
+        userID
+      )
+      await sbp('backend/publishLogEntry', attribute)
```

Comment:
This section (lines 49-54), needs to be moved up into the `/create` action, right after line 35, since this is part of the required setup that links the identity contract to the mailbox.